rs/cli/src/commands/update_authorized_subnets.rs

```diff
@@ -0,0 +1,158 @@
+
+        let human_bytes = human_bytes::human_bytes(self.state_size_limit as f64);
+        let agent = ctx.create_ic_agent_canister_client(None)?;
+
+        for subnet in subnets.values() {
+            if subnet.subnet_type.eq(&SubnetType::System) {
+                excluded_subnets.insert(subnet.principal.clone(), "System subnet".to_string());
+                continue;
+            }
+
+            let subnet_principal_string = subnet.principal.to_string();
+            if let Some((_, description)) = csv_contents.iter().find(|(short_id, _)| subnet_principal_string.starts_with(short_id)) {
+                excluded_subnets.insert(subnet.principal.clone(), description.to_owned());
+                continue;
+            }
+
+            let subnet_metrics = agent.read_state_subnet_metrics(&subnet.principal).await?;
+
+            if subnet_metrics.num_canisters >= self.canister_limit {
+                excluded_subnets.insert(
+                    subnet.principal.clone(),
+                    format!("Subnet has more than {} canisters", self.canister_limit),
+                );
+                continue;
+            }
+
+            if subnet_metrics.canister_state_bytes >= self.state_size_limit {
+                excluded_subnets.insert(subnet.principal.clone(), format!("Subnet has more than {} state size", human_bytes));
+            }
+        }
+
+        let summary = construct_summary(&subnets, &excluded_subnets)?;
+
+        let authorized = subnets
+            .keys()
+            .filter(|subnet_id| !excluded_subnets.contains_key(subnet_id))
+            .cloned()
+            .collect();
+
+        let ic_admin = ctx.ic_admin();
+        ic_admin
+            .propose_run(
+                ProposeCommand::SetAuthorizedSubnetworks { subnets: authorized },
+                ProposeOptions {
+                    title: Some("Update list of public subnets".to_string()),
+                    summary: Some(summary),
+                    motivation: None,
+                },
+            )
+            .await?;
+
+        Ok(())
+    }
+}
+
+impl UpdateAuthorizedSubnets {
+    fn parse_csv(&self) -> anyhow::Result<Vec<(String, String)>> {
+        let contents = BufReader::new(File::open(&self.path)?);
+        let mut ret = vec![];
+        for line in contents.lines() {
+            let content = line?;
+            if content.starts_with("subnet id") {
+                info!("Skipping header line in csv");
+                continue;
+            }
+
+            let (id, desc) = content.split_once(',').ok_or(anyhow::anyhow!("Failed to parse line: {}", content))?;
+            ret.push((id.to_string(), desc.to_string()))
+        }
+
+        Ok(ret)
+    }
+}
+
+fn construct_summary(subnets: &Arc<BTreeMap<PrincipalId, Subnet>>, excluded_subnets: &BTreeMap<PrincipalId, String>) -> anyhow::Result<String> {
+    Ok(format!(
+        "Updating the list of authorized subnets to:
+
+| Subnet id | Public | Description |
```

Comment:
We could have used `tabular` for this, since we already have it as a dependency?

Reply:
Doesn't tabular leave unnecessary whitespace for empty values? I think it was made for terminal, not for MD, but technically we can